quickwit-doc-mapper/src/default_doc_mapper/default_mapper.rs

```diff
@@ -369,42 +83,39 @@ impl From<DefaultDocMapper> for DefaultDocMapperBuilder {
 #[derive(Serialize, Deserialize, Clone)]
 #[serde(try_from = "DefaultDocMapperBuilder", into = "DefaultDocMapperBuilder")]
 pub struct DefaultDocMapper {
-    /// Store the json source in a text field _source.
-    pub store_source: bool,
+    /// Field in which the source should be stored.
+    pub(crate) source_field: Option<Field>,
```

Comment:
I was a bit surprised to see a tantivy `Field` as usually we don't want to store it directly, it's only relevant to the `DefaultDocMapper` schema and not for an external schema. I would avoid having putting a `pub` here and enrich a bit the docs `/// Field in which the source should be stored, can only be used with a `DefaultDocMapper` schema.`

Reply:
Good point.
I added the comment and made everything private.

I also opened #1340 to see if we can simplify the weird DocMapper contract.